app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -108,6 +114,12 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
 
   private val isNotConnected get() = conMan.activeNetworkInfo?.isConnected == false
 
+  private val isWriteStoragePermissionAllowed
+    get() = ContextCompat.checkSelfPermission(
+      requireActivity(),
+      Manifest.permission.WRITE_EXTERNAL_STORAGE
+    ) == PackageManager.PERMISSION_GRANTED
```

Comment:
We should also check if the user didn't give us permission and we also ask permission rationally, why do we need this permission, and maybe can we create an extension function here like this?
```
hasPermission(permission: String)
```

Reply:
done